src/components/pages/packages/flyouts/deletePackage/deletePackage.js

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React from 'react';
+
+import { LinkedComponent } from 'utilities';
+import {
+  AjaxError,
+  Btn,
+  BtnToolbar,
+  Indicator,
+  Modal
+} from 'components/shared';
+import { svgs } from 'utilities';
+
+import './deletePackage.css';
+
+export class DeletePackage extends LinkedComponent {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      changesApplied: false
+    };
+  }
+
+  apply = () => {
+    const { deletePackage, package: { id } } = this.props;
+    deletePackage(id);
+    this.setState({ changesApplied: true });
+  }
+
+  render() {
+    const { t, onClose, isPending, error } = this.props;
+    const { changesApplied } = this.state;
+    if (changesApplied && !error && !isPending) {
+      onClose();
+    }
+    return (
+      <Modal onClose={onClose} className="delete-package-container">
```

Comment:
Why is this using a modal (and introducing a new component for it)? Instead of using the already existing pattern (see device delete).

Reply:
This is the new design pattern given by Caitlin and everyone has agreed on this. So, it is too late to change it now.

Also, since it has very little content, It's fine to have it as a modal.

Reply:
Then there needs to be a corresponding bug/task to fix the existing ones, right?

Reply:
https://github.com/Azure/pcs-remote-monitoring-webui/issues/1080